apps/contrib/templatetags/custom_metadata_tags.py

```diff
@@ -18,11 +17,9 @@ def custom_meta_tags(context, model=None):
         return ""
     if not model:
         model = context.get('self', None)
-        if not hasattr(model, 'get_meta_title') or (
-                not model.meta_title and
+        while not hasattr(model, 'get_meta_title') or (
+                not model.get_meta_title() and
                 not model.get_meta_description()):
-            site = Site.find_for_request(request)
-            if site:
-                model = site.root_page.specific
+            model = model.specific.get_parent().specific
```

Comment:
I tested an overview page without meta tags first and get an error here: `'NoneType' object has no attribute 'specific'`

Reply:
Uh, how weird, the overview should have the HomePage as parent, shouldn't it? Without the tags, I only tested the gurenbuch overview thiugh and that was fine. But I'll try all options, we have then!

Reply:
Oh right, or the overview itself didn't have the specific? Something weird. Good catch, though!

Reply:
I couldn't recreate that. In all versions: overview pages without child pages, with child pages without metatags and with child pages with meta tags. Could we have a look at your page on Tuesday?

(But I found another problem, I fixed. :))

Reply:
@Rineee This was a problem when the root page didn't have meta tags. It should work in any case now. :crossed_fingers:

Reply:
Ok, so now I only get that error on the homepage. So I do need the meta tags for the homepage, do I @fuzzylogic2000 ?

Reply:
Hmm, no, I will also catch that!

Reply:
Oh, I still had it the wrong way round. Now it's fine!!!!1!!!

Reply:
@Rineee Can you please do another round of reviewing?